src/zenml/models/v2/core/artifact_version.py

```diff
@@ -0,0 +1,392 @@
+    def visualizations(
+        self,
+    ) -> Optional[List["ArtifactVisualizationResponse"]]:
+        """The `visualizations` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_metadata().visualizations
+
+    @property
+    def run_metadata(self) -> Dict[str, "RunMetadataResponse"]:
+        """The `metadata` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_metadata().run_metadata
+
+    @property
+    def materializer(self) -> Source:
+        """The `materializer` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_metadata().materializer
+
+    @property
+    def data_type(self) -> Source:
+        """The `data_type` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.get_metadata().data_type
+
+    # Helper methods
+    @property
+    def name(self) -> str:
+        """The `name` property.
+
+        Returns:
+            the value of the property.
+        """
+        return self.artifact.name
+
+    @property
+    def step(self) -> "StepRunResponse":
+        """Get the step that produced this artifact.
+
+        Returns:
+            The step that produced this artifact.
+        """
+        from zenml.artifacts.utils import get_producer_step_of_artifact
+
+        return get_producer_step_of_artifact(self)
+
+    @property
+    def run(self) -> "PipelineRunResponse":
+        """Get the pipeline run that produced this artifact.
+
+        Returns:
+            The pipeline run that produced this artifact.
+        """
+        from zenml.client import Client
+
+        return Client().get_pipeline_run(self.step.pipeline_run_id)
+
+    def load(self) -> Any:
+        """Materializes (loads) the data stored in this artifact.
+
+        Returns:
+            The materialized data.
+        """
+        from zenml.artifacts.utils import load_artifact_from_response
+
+        return load_artifact_from_response(self)
+
+    def read(self) -> Any:
+        """(Deprecated) Materializes (loads) the data stored in this artifact.
+
+        Returns:
+            The materialized data.
+        """
+        logger.warning(
+            "`artifact.read()` is deprecated and will be removed in a future "
+            "release. Please use `artifact.load()` instead."
+        )
+        return self.load()
+
+    def visualize(self, title: Optional[str] = None) -> None:
+        """Visualize the artifact in notebook environments.
+
+        Args:
+            title: Optional title to show before the visualizations.
+        """
+        from zenml.utils.visualization_utils import visualize_artifact
+
+        visualize_artifact(self, title=title)
+
+
+# ------------------ Filter Model ------------------
+
+
+class ArtifactVersionFilter(WorkspaceScopedFilter):
+    """Model to enable advanced filtering of all Artifacts."""
```

Comment:
```suggestion
    """Model to enable advanced filtering of all artifacts."""
```

Since you lower-cased things elsewhere in this PR?